### PR TITLE
ExampleYoyo port

### PR DESCRIPTION
**Pull request by Falcon-05**
### What are the changes to ExampleMod?
Ports ExampleYoyo and its projectile to 1.4.

### CODE REVIEW

File: ExampleMod/Content/Items/Weapons/ExampleYoyo.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	public class ExampleYoyo : ModItem
+	{
+		public override void SetStaticDefaults() {
+			Tooltip.SetDefault("This is a modded yoyo.");
+			CreativeItemSacrificesCatalog.Instance.SacrificeCountNeededByItemId[Type] = 1; //The amount of this item that needs to be researched to unlock it in the journey mode duplication menu.
+
+			// These are all related to gamepad controls and don't seem to affect anything else
+			ItemID.Sets.Yoyo[Item.type] = true;
+			ItemID.Sets.GamepadExtraRange[Item.type] = 15;
+			ItemID.Sets.GamepadSmartQuickReach[Item.type] = true;
```
Comment (Steviegt6) [inline ExampleMod/Content/Items/Weapons/ExampleYoyo.cs:20]: Actual explanations for what these do would be useful. For example, what does `GamepadExtraRange` actually mean? What "range" does it extend?

File: ExampleMod/Content/Projectiles/ExampleYoyoProjectile.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+																		
+			ProjectileID.Sets.YoyosTopSpeed[Projectile.type] = 17f; //The top speed of the yoyo.
+			// Vanilla values range from 9f(Wood) to 17.5f(Terrarian), and defaults to 10f.
+		}
+
+		public override void SetDefaults() {
+			Projectile.width = 16; //The width of the projectile's hitbox.
+			Projectile.height = 16; //The height of the projectile's hitbox.
+
+			Projectile.aiStyle = 99; //The projectile's ai style. Yoyos use aiStyle 99.
```
Comment (Steviegt6) [inline ExampleMod/Content/Projectiles/ExampleYoyoProjectile.cs:26]: Don't we have AIStyleIDs in 1.4? I forget.

**Review by steviegt6 — COMMENTED** [review]
Not exactly asking you to take care of this, but this is just my input.

File: ExampleMod/Content/Items/Weapons/ExampleYoyo.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+			Item.useStyle = ItemUseStyleID.Shoot; // The way the item is used (e.g. swinging, throwing, etc.)
+			Item.useTime = 25; // The item's use time in ticks (1 second == 60 ticks)
+			Item.useAnimation = 25; // The item's use animation length in ticks (1 second == 60 ticks)
+			Item.noMelee = true; // The item's animation won't deal damage if this is set to true.
+			Item.noUseGraphic = true; // The item's animation will be invisible if this is set to true.
+			Item.UseSound = SoundID.Item1; // The sound that will play when the item is used.
+
+			Item.damage = 40; // The amount of damage the item does to an enemy or player.
+			Item.DamageType = DamageClass.Melee; // Sets the item's damage type to melee.
```
Comment (Rijam) [inline ExampleMod/Content/Items/Weapons/ExampleYoyo.cs]: This should be changed to `DamageClass.MeleeNoSpeed` now.

**Comment by JavidPack** [comment]
Thanks a bunch, sorry this tool so long.

**Pull request merged**